api/utils/prompt/stdin.go

```diff
@@ -69,3 +74,36 @@ func NotifyExit() {
 	}
 	stdinMU.Unlock()
 }
+
+// EnableStdinTerminalFallback attempts to use a fallback like /dev/tty when stdin is
+// not the terminal. This can be useful when tsh is called by another terminal
+// tool (e.g. git).
+func EnableStdinTerminalFallback() {
+	stdinMU.Lock()
+	defer stdinMU.Unlock()
+	stdinTerminalFallback = true
+}
+
+func maybeUseStdinTerminalFallbackLocked(original *ContextReader) *ContextReader {
+	if !stdinTerminalFallback || runtime.GOOS == "windows" || original.IsTerminal() {
+		return original
+	}
+
+	// File /dev/tty is the controlling tty of the current terminal. Not
+	// available on Windows.
+	// https://tldp.org/HOWTO/Text-Terminal-HOWTO-7.html
+	devTTY, err := os.Open("/dev/tty")
+	if err != nil {
+		slog.DebugContext(context.Background(), "Failed to open /dev/tty", "error", err)
```

Comment:
Nit: Perhaps define `ctx := context.Background()` once at the start of the function?

Reply:
https://github.com/gravitational/teleport/pull/51305/commits/8a33dbba1caed70ef96cf29384ed543ba39e0084